### ISSUE TITLE
Patch for using with the latest MessagePack for C++ (1.1.0)

### ISSUE THREAD

**Issue by ducss2015** [jubatus/jubatus-msgpack-rpc#20]
I created a patch to compile jubatus-msgpack-rpc for cpp with the latest [MessagePack for C++ version 1.1.0](https://github.com/msgpack/msgpack-c), as following. I am not sure if there is any other incompatibility issues or not. Please review it if you have some time.

```
diff --git a/src/jubatus/msgpack/rpc/exception.cc b/src/jubatus/msgpack/rpc/exception.cc
index 5fd60da..afadcc1 100644
--- a/src/jubatus/msgpack/rpc/exception.cc
+++ b/src/jubatus/msgpack/rpc/exception.cc
@@ -56,20 +56,20 @@ void throw_exception(future_impl* f)
         //   caused by *local* errors. Otherwise, caused by
         //   *remote* errors or delivered errors from remote.

- if(err.type == msgpack::type::RAW &&
-     err.via.raw.ptr == TIMEOUT_ERROR_PTR) {
+ if(err.type == msgpack::type::BIN &&
+     err.via.bin.ptr == TIMEOUT_ERROR_PTR) {
    throw timeout_error();

- } else if(err.type == msgpack::type::RAW &&
-     err.via.raw.ptr == CONNECT_ERROR_PTR) {
+ } else if(err.type == msgpack::type::BIN &&
+     err.via.bin.ptr == CONNECT_ERROR_PTR) {
    throw connect_error();

-        } else if(err.type == msgpack::type::RAW &&
-     err.via.raw.ptr == REQUEST_CANCELLED_PTR) {
+        } else if(err.type == msgpack::type::BIN &&
+     err.via.bin.ptr == REQUEST_CANCELLED_PTR) {
    throw request_cancelled();

-        } else if(err.type == msgpack::type::RAW &&
-     err.via.raw.ptr == CONNECTION_CLOSED_ERROR_PTR) {
+        } else if(err.type == msgpack::type::BIN &&
+     err.via.bin.ptr == CONNECTION_CLOSED_ERROR_PTR) {
    throw connection_closed_error();

         } else if(err.type == msgpack::type::NEGATIVE_INTEGER ) {
@@ -87,11 +87,11 @@ void throw_exception(future_impl* f)
      err.via.u64 == ARGUMENT_ERROR) {
    throw argument_error();

-        } else if ( err.type == msgpack::type::RAW ) {
+        } else if ( err.type == msgpack::type::BIN ) {
           // maybe simple error message

           std::ostringstream os;
-          os.write(err.via.raw.ptr, err.via.raw.size);
+          os.write(err.via.bin.ptr, err.via.bin.size);
           throw remote_error(os.str(), future(f->shared_from_this()));

  } else {
diff --git a/src/jubatus/msgpack/rpc/transport/base.h b/src/jubatus/msgpack/rpc/transport/base.h
index e1036a5..468dbd5 100644
--- a/src/jubatus/msgpack/rpc/transport/base.h
+++ b/src/jubatus/msgpack/rpc/transport/base.h
@@ -144,8 +144,9 @@ private:
 template <typename MixIn>
 inline stream_handler<MixIn>::stream_handler(int fd, loop lo) :
  mp::wavy::handler(fd),
- m_pac(MSGPACK_RPC_STREAM_BUFFER_SIZE),
- m_loop(lo) { }
+ m_loop(lo) {
+ m_pac.reserve_buffer(MSGPACK_RPC_STREAM_BUFFER_SIZE);
+}

 template <typename MixIn>
 inline stream_handler<MixIn>::~stream_handler() { }
```

**Comment by ducss2015** [comment]
The following is the patch for the tests. In this patch, I changed the call to msgpack::zone::malloc() to msgpack::zone::allocate_no_align()
```
diff --git a/test/attack_callback.cc b/test/attack_callback.cc
index 5f4c83a..24a9100 100644
--- a/test/attack_callback.cc
+++ b/test/attack_callback.cc
@@ -37,7 +37,7 @@ void attack_callback()
    s.set_timeout(30.0);

    rpc::shared_zone msglife(new msgpack::zone());
-   raw_ref msg = raw_ref((char*)msglife->malloc(ATTACK_SIZE), ATTACK_SIZE);
+   raw_ref msg = raw_ref((char*)msglife->allocate_no_align(ATTACK_SIZE), ATTACK_SIZE);

    for(size_t j=0; j < ATTACK_DEPTH; ++j) {
      pipeline[j] = s.call("echo_huge", msglife, msg).attach_callback(
diff --git a/test/attack_huge.cc b/test/attack_huge.cc
index 87fac14..d61b075 100644
--- a/test/attack_huge.cc
+++ b/test/attack_huge.cc
@@ -45,7 +45,7 @@ int main(void)
    << std::endl;

  msglife.reset(new msgpack::zone());
- msg = raw_ref((char*)msglife->malloc(ATTACK_SIZE), ATTACK_SIZE);
+ msg = raw_ref((char*)msglife->allocate_no_align(ATTACK_SIZE), ATTACK_SIZE);

  test.reset(new attacker());
  test->run(ATTACK_THREAD, &attack_huge);
```

**Comment by kmaehashi** [comment]
Thanks for contribution!
We decided to maintain the MessagePack-1.x compatible code in branch named `msgpack-1.x`.
Could you please make a pull-request to that branch?

**Comment by kmaehashi** [comment]
I manually merged into `msgpack-1.1` branch. Thanks again for contribution!
